src/Core/src/Platform/iOS/ButtonExtensions.cs

```diff
@@ -59,13 +61,28 @@ public static void UpdateFont(this UIButton nativeButton, ITextStyle textStyle,
 			nativeButton.TitleLabel.UpdateFont(textStyle, fontManager, UIFont.ButtonFontSize);
 		}
 
-		public static void UpdatePadding(this UIButton nativeButton, IButton button)
+		public static void UpdatePadding(this UIButton nativeButton, IButton button, Thickness? defaultPadding = null) =>
+			UpdatePadding(nativeButton, button.Padding, defaultPadding);
+
+		public static void UpdatePadding(this UIButton nativeButton, Thickness padding, Thickness? defaultPadding = null)
 		{
+			if (padding.IsNaN)
+				padding = defaultPadding ?? Thickness.Zero;
+
+			// top and bottom insets reset to a "default" if they are exactly 0
+			// however, internally they are floor-ed, so there is no actual fractions
+			var top = padding.Top;
+			if (top == 0.0)
+				top = AlmostZero;
+			var bottom = padding.Bottom;
+			if (bottom == 0.0)
+				bottom = AlmostZero;
```

Comment:
Just when you think "iOS has no padding, so it will be easy", Apple throws you a curve ball - or a rotten apple if you like. When you set padding to 0, iOS decides that you are wrong and will use the 0 for the left/right padding, but a 6 or 8 padding for top or bottom. So, to avoid this, we have to set a "almost zero" padding for the top/bottom.

But no worries, iOS does not respect this fraction as it seems to trim the value. So, not only can you not set 0, but it also doesn't respect half the non-zero values!